source/fundamentals/connection.txt

```diff
@@ -0,0 +1,636 @@
+       response time plus the local threshold, in milliseconds.
+
+   * - **heartbeatFrequencyMS**
+     - integer
+     - Specifies the frequency, in milliseconds, that the driver will
+       wait between attempts to determine the current state of each
+       server in the cluster.
+
+   * - **replicaSet**
+     - string
+     - Specifies that the :ref:`connection string <connection-uri>`
+       provided includes multiple hosts. When specified, the Java driver
+       attempts to find all members of that set.
+
+   * - **ssl**
+     - boolean
+     - Specifies that all communication with MongoDB instances should
+       use TLS.
+
+   * - **tls**
+     - boolean
+     - Specifies that all communication with MongoDB instances should
+       use TLS. Supersedes the **ssl** option.
+
+   * - **tlsInsecure**
+     - boolean
+     - Specifies that the driver should allow invalid host names for TLS
+       connections. Has the same effect as setting
+       **tlsAllowInvalidHostnames** to true. To relax TLS security
+       constraints in other ways, use a
+       :ref:`custom SSLContext <tls-custom-sslContext>`.
+
+   * - **sslInvalidHostNameAllowed**
+     - boolean
+     - Specifies that the driver should allow invalid host names for TLS
+       connections.
+
+   * - **tlsAllowInvalidHostnames**
+     - boolean
+     - Specifies that the driver should allow invalid host names for TLS
+       connections. Supersedes **sslInvalidHostNameAllowed**.
+
+   * - **connectTimeoutMS**
+     - integer
+     - Specifies the maximum amount of time, in milliseconds, the Java
+       driver will wait for a connection to open before timing out.
+
+   * - **socketTimeoutMS**
+     - integer
+     - Specifies the maximum amount of time, in milliseconds, the Java
+       driver will wait to send or receive a request before timing out.
+
+   * - **maxIdleTimeMS**
+     - integer
+     - Specifies the maximum amount of time, in milliseconds, the Java
+       driver will allow a pooled connection to idle before closing the
+       connection.
+
+   * - **maxLifeTimeMS**
+     - integer
+     - Specifies the maximum amount of time, in milliseconds, the Java
+       driver will continue to use a pooled connection before closing the
+       connection.
+
+   * - **safe**
+     - boolean
+     - Specifies that the driver ensures that all writes are acknowledged
+       by the connected MongoDB instance. Throws an exception on any write
+       that does not receive acknowledgement from the connected MongoDB
+       instance.
+
+   * - **journal**
+     - boolean
+     - Specifies that the driver must wait for the connected MongoDB
+       instance to group commit to the journal file on disk for all writes.
+
+   * - **w**
+     - string or integer
+     - Specifies the write concern. For more information on values, see
+       the server documentation for the :manual:`w option
+       </reference/write-concern/#w-option>`. Implies **safe=true**.
```

Comment:
Question: what does "Implies safe=true" mean?

Reply:
Reworded to make this clearer; let me know if you have any further questions.